assignments/arrays.js

```diff
@@ -1,96 +1,151 @@
 let lastCar = 0;
-console.log();
+for (let i=inventory.length-1; i<inventory.length; i++) {
+    console.log(`Last car in inventory is ID#${inventory[i].id} \n${inventory[i].car_make} ${inventory[i].car_model} \n-------------------------------\n`)
+}
+
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+
+// Copy inventory array of objects into a new variable: carModels then sort in place by 'car_model' using the above sortOn function 
 let carModels = [];
-console.log();
+carModels = inventory.slice();
+sortOn(carModels, 'car_model');
+
+// Iterate over the array logging results in console
+for (let i=0; i<carModels.length; i++) {
+    console.log(carModels[i].car_model);
+}
+console.log('--- Car models sorted alphabetically ---\n-------------------------------\n');
+
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+
+// Copy the inventory array of objects into carYears then sort using sortOn by 'car_year' (no need to sort, according to question, but looks better)
+let carYears = []; // First create empty array
+carYears = inventory.slice(); // this is how to make a copy of existing array into new empty array
+sortOn(carYears, 'car_year');
+
+// Iterate over array and create a new array of just car_year
+for (let i=0; i<carYears.length; i++) {
+    console.log(carYears[i].car_year);
+}
+console.log('--- Car years sorted numerically ---\n-------------------------------\n');
+
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+
+// Create a new, empty array called oldCars
+let oldCars = [];
+
+// Loop over the carYears array and pick out all carYears before 2000. Push those values into the new array, oldCars
+for (let i = 0; i < carYears.length; i++) {
+    if (carYears[i].car_year < 2000) {
+        oldCars.push(carYears[i].car_year);
+    }
+}
+console.log(`There are ${oldCars.length} total cars in inventory made before 2000\n-------------------------------\n`);
+
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
-let BMWAndAudi =[];
-console.log();
-
 
+let BMWAndAudi = [];
+for (let i = 0; i < inventory.length; i++) {
+    if ( inventory[i].car_make === 'BMW' ) {
```

Comment:
Consider just using the `||` (or) operand in your if statement. Then you wouldn't need the else portion of this, much shorter and cleaner.
```suggestion
    if ( inventory[i].car_make === 'BMW' || inventory[i].car_make === 'Audi') {
```